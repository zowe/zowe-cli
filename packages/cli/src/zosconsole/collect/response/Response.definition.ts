/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { ICommandDefinition } from "@zowe/imperative";
import { Constants } from "../../../Constants";

export const SyncResponseCommandDefinition: ICommandDefinition = {
    name: "sync-responses",
    aliases: ["sr"],
    summary: "Collect outstanding synchronous console response messages",
    description: `The z/OSMF console REST APIs return a "solicited response key" after successfully issuing ` +
    `a synchronous console command that produces solicited responses. You can use the "solicited response key"` +
    `on the "sync-responses" command to collect any additional outstanding solicited responses ` +
    `from the console the command was issued.` +
    `\n\n` +
    `In general, when issuing a z/OS console command, z/OS applications route responses to the originating console. ` +
    `The command response messages are referred to as "solicited command responses" (i.e. direct responses to the command issued). ` +
    `When issuing a z/OS console command using ${Constants.DISPLAY_NAME}, collection of all solicited command responses ` +
    `is attempted by default. However, there is no z/OS mechanism that indicates the total number of response messages that may be produced ` +
    `from a given command. Therefore, the ${Constants.DISPLAY_NAME} console APIs return a "solicited response key" ` +
    `that can be used to "follow-up" and collect any additional solicited command responses.`,
    type: "command",
    handler: __dirname + "/Response.handler",
    profile: {
        optional: ["zosmf"]
    },
    positionals: [
        {
            name: "responsekey",
            type: "string",
            description: `The "solicited response key" provided in response to a previously issued console command. ` +
            `Used by the z/OSMF console API to collect any additional outstanding solicited responses from a ` +
            `previously issued console command.`,
            regex: "^[a-zA-Z0-9]+$",
            required: true
        }
    ],
    options: [
        {
            name: "console-name",
            aliases: ["cn", "c"],
            description: `The name of the z/OS extended MCS console to direct the command. ` +
            `You must have the required authority to access the console specified. ` +
            `You may also specify an arbitrary name, if your installation allows dynamic creation ` +
            `of consoles with arbitrary names.`,
            type: "string",
            allowableValues: {
                values: [/^[a-zA-Z0-9]+$/.source],
                caseSensitive: false
            }
        }
    ],
    examples: [
        {
            description: "Collect any outstanding additional solicited response messages",
            options: "C4866969"
        }
    ]
};
