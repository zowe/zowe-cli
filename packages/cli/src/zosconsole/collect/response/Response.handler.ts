/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { IHandlerParameters } from "@zowe/imperative";
import { CollectCommand, ICollectParms, IConsoleResponse } from "@zowe/zos-console-for-zowe-sdk";
import { ZosmfBaseHandler } from "@zowe/zosmf-for-zowe-sdk";

/**
 * Handle to collect a MVS console command response
 * @export
 * @class Handler
 * @implements {ICommandHandler}
 */
export default class Handler extends ZosmfBaseHandler {

    public async processCmd(commandParameters: IHandlerParameters) {

        const collectParms: ICollectParms = {
            commandResponseKey: commandParameters.arguments.responsekey,
            consoleName: commandParameters.arguments["console-name"]
        };

        const response: IConsoleResponse = await CollectCommand.collect(this.mSession, collectParms);

        // Print out the response
        commandParameters.response.console.log(response.commandResponse);

        // Return as an object when using --response-format-json
        commandParameters.response.data.setObj(response);

    }
}
