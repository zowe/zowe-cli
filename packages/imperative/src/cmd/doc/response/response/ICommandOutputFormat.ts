/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

/**
 * The available format types for formatting of output.
 *
 * list
 * ====
 * Formats the output data as a list of strings. If an array of objects is present in the output, each is stringified
 * and printed on a newline.
 *
 * table
 * =====
 * Formats the output data as a table using the properties as the column headers. You must ensure that the array of
 * objects is homogeneous for the table to print properly.
 *
 * string
 * ======
 * Formats the output data as a string. If the output data is an object/array it is stringified.
 *
 * object
 * ======
 * Formats the output a prettified JSON object.
 *
 * @type {OUTPUT_FORMAT}
 */
export type OUTPUT_FORMAT = "list" | "table" | "string" | "object" ;

/**
 * The output format object is returned from a successful command handler to dictate how the output for the
 * command should be formatted. This is an optional feature for Imperative and handlers are not required to
 * return this object (they can format their own output).
 *
 * These properties represent the defaults for the handler. If the command definition includes the output format
 * options, the options take precedence over the default values (see ICommandDefinition for details).
 * @export
 * @interface ICommandOutput
 */
export interface ICommandOutputFormat {
    /**
     * The output data to format. Common output data includes arrays of strings/objects, JSON objects, strings.
     * @type {*}
     * @memberof ICommandOutputFormat
     */
    output: any;
    /**
     * The output format type (see the type for details). In most cases, any data type returned (on the output property)
     * can be formatted according to the type specified here.
     * @type {OUTPUT_FORMAT}
     * @memberof ICommandOutput
     */
    format: OUTPUT_FORMAT;
    /**
     * If the response is an object (or an array of objects) these are the top level properties to keep. For example,
     * if an array of homogeneous objects are returned in the output, the fields that are NOT specified in this array
     * are removed/deleted from each object.
     * @type {string[]}
     * @memberof ICommandOutput
     */
    fields?: string[];
    /**
     * If response format table is specified, print the table headings
     * @type {boolean}
     * @memberof ICommandOutput
     */
    header?: boolean;
}
