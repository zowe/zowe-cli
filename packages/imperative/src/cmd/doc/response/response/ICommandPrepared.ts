/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { CommandProfiles } from "../../../profiles/CommandProfiles";
import { ICommandArguments } from "../../../doc/args/ICommandArguments";
/**
 * Command Processor prepare response.
 * @export
 * @interface ICommandPrepared
 */
export interface ICommandPrepared {
    /**
     * The profile map object for all profiles loaded for commands.
     * @type {CommandProfiles}
     * @memberof ICommandPrepared
     */
    profiles: CommandProfiles;
    /**
     * Imperative arguments object. Starts with arguments passed parsed by
     * Yargs as a base and fills in the rest from ENV/profile/defaults.
     * Eventually passed to handlers.
     * @type {ICommandArguments}
     */
    args: ICommandArguments;
}
