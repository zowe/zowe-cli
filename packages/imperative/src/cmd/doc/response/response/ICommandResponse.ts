/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { IImperativeError } from "../../../../error";

/**
 * Command response object built by the command processor (and command handler). The response object is always
 * built internally, but displayed to the command issuer if response-format-json is specified.
 *
 * For handlers, see the HandlerResponse (IHandlerResponseApi) for the public APIs used to by handlers to create
 * this response.
 * @export
 * @interface ICommandResponse
 */
export interface ICommandResponse {
    /**
     * Overall command success flag. True indicates that the command handler/processor/help was successful.
     * @type {boolean}
     * @memberof ICommandResponse
     */
    success: boolean;
    /**
     * Requested exit code for the process when your command is complete.
     * If this is not specified, the default is 0 for successful commands and 1 for failed commands
     * according to the value of the above "success" field.
     * @type {number}
     * @memberof ICommandResponse
     */
    exitCode: number;
    /**
     * Message appended by the handlers. The message is not displayed on the console, only displayed if response format
     * JSON is indicated.
     * @type {string}
     * @memberof ICommandResponse
     */
    message: string;
    /**
     * The stdout from the command. Buffered regardless of response format specification.
     * @type {Buffer}
     * @memberof ICommandResponse
     */
    stdout?: Buffer;
    /**
     * The stderr from the command. Buffered regardless of response format specification.
     * @type {Buffer}
     * @memberof ICommandResponse
     */
    stderr?: Buffer;
    /**
     * Handlers (and help, etc.) can choose to append a data object to the response. Not displayed to the console
     * unless response format JSON is specified.
     * @type {*}
     * @memberof ICommandResponse
     */
    data?: any;
    /**
     * Error object automatically appended by the command processor when a handler rejects the promise. Contains the
     * stack and other messages to help diagnosis. Not displayed to the console unless response format JSON is specified.
     * @type {IImperativeError}
     * @memberof ICommandResponse
     */
    error?: IImperativeError;
}
