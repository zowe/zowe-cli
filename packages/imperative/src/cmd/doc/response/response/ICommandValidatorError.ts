/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

/**
 * Describes the operand in error and provides the full definition for the option/operand - normally exposed when the
 * JSON response format is requested.
 * @export
 * @interface ICommandValidatorError
 */
export interface ICommandValidatorError {
    /**
     * The validation error message.
     * @type {string}
     * @memberof ICommandValidatorError
     */
    message: string;
    /**
     * The option that failed validation.
     * @type {string}
     * @memberof ICommandValidatorError
     */
    optionInError?: string;
    /**
     * The option definition that failed.
     * @type {*}
     * @memberof ICommandValidatorError
     */
    definition?: any;
}
