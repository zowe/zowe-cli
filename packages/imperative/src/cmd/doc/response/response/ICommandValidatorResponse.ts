/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

/**
 * Syntax validator response.
 * TODO - In the future, when we supply an the ability to override/extend the SyntaxValidator, we should change the
 * TODO - validator from printing the syntax errors itself to returning an object with a structured error list.
 * @export
 * @interface ICommandValidatorResponse
 */
export interface ICommandValidatorResponse {
    /**
     * Indicates if the syntax/parameters supplied by the user were valid.
     * @type {boolean}
     * @memberof ICommandValidatorResponse
     */
    valid: boolean;
}
