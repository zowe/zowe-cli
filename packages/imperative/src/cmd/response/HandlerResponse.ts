/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { CommandResponse } from "./CommandResponse";
import { IHandlerResponseDataApi } from "../doc/response/api/handler/IHandlerResponseDataApi";
import { IHandlerResponseConsoleApi } from "../doc/response/api/handler/IHandlerResponseConsoleApi";
import { IHandlerResponseApi } from "../doc/response/api/handler/IHandlerResponseApi";
import { IHandlerProgressApi } from "../doc/response/api/handler/IHandlerProgressApi";
import { IHandlerFormatOutputApi } from "../doc/response/api/handler/IHandlerFormatOutputApi";

/**
 * Handler command response APIs. Passed to command handlers for console messages, data responses, and progress bars.
 * Handlers can implement their own logging scheme (or use the Imperative.api to obtain an instance of the logger),
 * but handlers should NEVER write directly to the console and should always use the APIs here to construct responses.
 * The HandlerResponse object will automatically handle response formats, etc.
 * @export
 * @class HandlerResponse
 * @implements {IHandlerResponseApi}
 */
export class HandlerResponse implements IHandlerResponseApi {
    /**
     * The command response object allocated by the command processor that exposes the APIs intended for the handler
     * to use.
     * @private
     * @type {CommandResponse}
     * @memberof HandlerResponse
     */
    private mApis: CommandResponse;
    /**
     * Creates an instance of HandlerResponse.
     * @param {CommandResponse} apis - The command response object (See class for details) - contains all the public
     * APIs that the handler should use to formulate responses.
     * @memberof HandlerResponse
     */
    constructor(apis: CommandResponse) {
        this.mApis = apis;
    }
    /**
     * Returns the instance of the console APIs for the handler. Provides APIs to console messaging (log, error,
     * etc.).
     * @readonly
     * @type {IHandlerResponseConsoleApi} - See the interface for API details.
     * @memberof HandlerResponse
     */
    public get console(): IHandlerResponseConsoleApi {
        return this.mApis.console;
    }
    /**
     * Returns the instance of the data APIs for the handler. Provides APIs for the handlers to formulate
     * their JSON response objects to the command.
     * @readonly
     * @type {IHandlerResponseDataApi} - See the interface for API details.
     * @memberof HandlerResponse
     */
    public get data(): IHandlerResponseDataApi {
        return this.mApis.data;
    }
    /**
     * Returns the instance of the progress bar APIs for the handler. Provides methods to start and end progress bars.
     * @readonly
     * @type {IHandlerProgressApi} - See the interface for API details.
     * @memberof HandlerResponse
     */
    public get progress(): IHandlerProgressApi {
        return this.mApis.progress;
    }

    /**
     * Returns the instance of the format APIs for the handler. APIs allow the handler to provide output objects/strings
     * and they will be automatically formatted and printed to the console.
     * @readonly
     * @type {IHandlerFormatOutputApi}
     * @memberof HandlerResponse
     */
    public get format(): IHandlerFormatOutputApi {
        return this.mApis.format;
    }
}
