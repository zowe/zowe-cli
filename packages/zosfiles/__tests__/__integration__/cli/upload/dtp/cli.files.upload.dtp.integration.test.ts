/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import * as path from "path";
import { runCliScript, stripNewLines } from "../../../../../../../__tests__/__src__/TestUtils";
import { TestEnvironment } from "../../../../../../../__tests__/__src__/environment/TestEnvironment";
import { ITestEnvironment } from "../../../../../../../__tests__/__src__/environment/doc/response/ITestEnvironment";

let TEST_ENVIRONMENT: ITestEnvironment;

describe("Upload directory to PDS", () => {

    beforeAll(async () => {

        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            testName: "upload_data_set",
            skipProperties: true
        });
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    it("should display the help", async () => {
        const shellScript = path.join(__dirname, "__scripts__", "command_upload_dtp_help.sh");
        const response = runCliScript(shellScript, TEST_ENVIRONMENT);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        const helpText = response.stdout.toString();
        expect(helpText).toMatchSnapshot();
    });

    describe("Expected failures", () => {
        it("should fail due to missing local directory name", async () => {
            const shellScript = path.join(__dirname, "__scripts__", "command", "command_upload_dtp.sh");
            const response = runCliScript(shellScript, TEST_ENVIRONMENT, [""]);
            expect(response.status).toBe(1);
            expect(response.stderr.toString()).toContain("Missing Positional Argument");
            expect(response.stderr.toString()).toContain("inputdir");
            expect(response.stderr.toString()).toContain("dataSetName");
        });

        it("should fail due to missing mf data set name", async () => {
            const shellScript = path.join(__dirname, "__scripts__", "command", "command_upload_dtp.sh");
            const localDirName = path.join(__dirname, "__data__", "command_upload_dtp_dir");
            const response = runCliScript(shellScript, TEST_ENVIRONMENT, [localDirName]);
            expect(response.status).toBe(1);
            expect(response.stderr.toString()).toContain("Missing Positional Argument");
            expect(response.stderr.toString()).toContain("dataSetName");
        });

        it("should fail when local directory does not exist", async () => {
            const shellScript = path.join(__dirname, "__scripts__", "command", "command_upload_dtp.sh");
            const response = runCliScript(shellScript, TEST_ENVIRONMENT, ["localDirThatDoesNotExist", "mf.data.set"]);
            expect(stripNewLines(response.stderr.toString())).toContain("no such file or directory, lstat");
            expect(stripNewLines(response.stderr.toString())).toContain("localDirThatDoesNotExist");
        });

    });
});

