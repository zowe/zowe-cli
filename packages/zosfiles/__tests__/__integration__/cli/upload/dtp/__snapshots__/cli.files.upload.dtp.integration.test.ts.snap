// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload directory to PDS should display the help 1`] = `
"================ Z/OS FILES UPLOAD DIRECTORY HELP ===============

 COMMAND NAME
 ------------

   dir-to-pds | dtp

 DESCRIPTION
 -----------

   Upload files from a directory to a partioned data set (PDS)

 USAGE
 -----

   zowe zos-files upload dir-to-pds <inputdir> <dataSetName> [options]

 POSITIONAL ARGUMENTS
 --------------------

   inputdir		 (string)

      The path for a local directory that you want to upload to a PDS

   dataSetName		 (string)

      The name of the partitioned data set to which you want to upload the files

 OPTIONS
 -------

   --binary  | -b (boolean)

      Data content in binary mode, which means that no data conversion is performed.
      The data transfer process returns each record as-is, without translation. No
      delimiters are added between records.

   --migrated-recall  | --mr (string)

      The method by which migrated data set is handled. By default, a migrated data
      set is recalled synchronously. You can specify the following values: wait,
      nowait, error

      Default value: nowait

   --volume-serial  | --vs (string)

      The volume serial (VOLSER) where the data set resides. You can use this option
      at any time. However, the VOLSER is required only when the data set is not
      cataloged on the system. A VOLSER is analogous to a drive name on a PC.

 ZOSMF CONNECTION OPTIONS
 ------------------------

   --host  | -H (string)

      The z/OSMF server host name.

   --port  | -P (number)

      The z/OSMF server port.

      Default value: 443

   --user  | -u (string)

      Mainframe (z/OSMF) user name, which can be the same as your TSO login.

   --password  | --pass | --pw (string)

      Mainframe (z/OSMF) password, which can be the same as your TSO password.

   --reject-unauthorized  | --ru (boolean)

      Reject self-signed certificates.

      Default value: true

   --base-path  | --bp (string)

      The base path for your API mediation layer instance. Specify this option to
      prepend the base path to all z/OSMF resources when making REST requests. Do not
      specify this option if you are not using an API mediation layer.

 PROFILE OPTIONS
 ---------------

   --zosmf-profile  | --zosmf-p (string)

      The name of a (zosmf) profile to load for this command execution.

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text


 EXAMPLES
 --------

   - Upload a directory named \\"src\\" to a PDS named
   \\"ibmuser.src\\":

      $ zowe zos-files upload dir-to-pds \\"src\\" \\"ibmuser.src\\"

   - Upload a directory named \\"src\\" to a migrated PDS named
   \\"ibmuser.src\\" and wait for it to be recalled:

      $ zowe zos-files upload dir-to-pds \\"src\\" \\"ibmuser.src\\" --mr wait

================ Z/OS FILES UPLOAD DIRECTORY HELP WITH RFJ ===========
{
  \\"success\\": true,
  \\"message\\": \\"The help was constructed for command: dir-to-pds.\\",
  \\"stdout\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   dir-to-pds | dtp\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Upload files from a directory to a partioned data set (PDS)\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe zos-files upload dir-to-pds <inputdir> <dataSetName> [options]\\\\n\\\\n POSITIONAL ARGUMENTS\\\\n --------------------\\\\n\\\\n   inputdir\\\\t\\\\t (string)\\\\n\\\\n      The path for a local directory that you want to upload to a PDS\\\\n\\\\n   dataSetName\\\\t\\\\t (string)\\\\n\\\\n      The name of the partitioned data set to which you want to upload the files\\\\n\\\\n OPTIONS\\\\n -------\\\\n\\\\n   --binary  | -b (boolean)\\\\n\\\\n      Data content in binary mode, which means that no data conversion is performed.\\\\n      The data transfer process returns each record as-is, without translation. No\\\\n      delimiters are added between records.\\\\n\\\\n   --migrated-recall  | --mr (string)\\\\n\\\\n      The method by which migrated data set is handled. By default, a migrated data\\\\n      set is recalled synchronously. You can specify the following values: wait,\\\\n      nowait, error\\\\n\\\\n      Default value: nowait\\\\n\\\\n   --volume-serial  | --vs (string)\\\\n\\\\n      The volume serial (VOLSER) where the data set resides. You can use this option\\\\n      at any time. However, the VOLSER is required only when the data set is not\\\\n      cataloged on the system. A VOLSER is analogous to a drive name on a PC.\\\\n\\\\n ZOSMF CONNECTION OPTIONS\\\\n ------------------------\\\\n\\\\n   --host  | -H (string)\\\\n\\\\n      The z/OSMF server host name.\\\\n\\\\n   --port  | -P (number)\\\\n\\\\n      The z/OSMF server port.\\\\n\\\\n      Default value: 443\\\\n\\\\n   --user  | -u (string)\\\\n\\\\n      Mainframe (z/OSMF) user name, which can be the same as your TSO login.\\\\n\\\\n   --password  | --pass | --pw (string)\\\\n\\\\n      Mainframe (z/OSMF) password, which can be the same as your TSO password.\\\\n\\\\n   --reject-unauthorized  | --ru (boolean)\\\\n\\\\n      Reject self-signed certificates.\\\\n\\\\n      Default value: true\\\\n\\\\n   --base-path  | --bp (string)\\\\n\\\\n      The base path for your API mediation layer instance. Specify this option to\\\\n      prepend the base path to all z/OSMF resources when making REST requests. Do not\\\\n      specify this option if you are not using an API mediation layer.\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --zosmf-profile  | --zosmf-p (string)\\\\n\\\\n      The name of a (zosmf) profile to load for this command execution.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   - Upload a directory named \\\\\\"src\\\\\\" to a PDS named\\\\n   \\\\\\"ibmuser.src\\\\\\":\\\\n\\\\n      $ zowe zos-files upload dir-to-pds \\\\\\"src\\\\\\" \\\\\\"ibmuser.src\\\\\\"\\\\n\\\\n   - Upload a directory named \\\\\\"src\\\\\\" to a migrated PDS named\\\\n   \\\\\\"ibmuser.src\\\\\\" and wait for it to be recalled:\\\\n\\\\n      $ zowe zos-files upload dir-to-pds \\\\\\"src\\\\\\" \\\\\\"ibmuser.src\\\\\\" --mr wait\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   dir-to-pds | dtp\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Upload files from a directory to a partioned data set (PDS)\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe zos-files upload dir-to-pds <inputdir> <dataSetName> [options]\\\\n\\\\n POSITIONAL ARGUMENTS\\\\n --------------------\\\\n\\\\n   inputdir\\\\t\\\\t (string)\\\\n\\\\n      The path for a local directory that you want to upload to a PDS\\\\n\\\\n   dataSetName\\\\t\\\\t (string)\\\\n\\\\n      The name of the partitioned data set to which you want to upload the files\\\\n\\\\n OPTIONS\\\\n -------\\\\n\\\\n   --binary  | -b (boolean)\\\\n\\\\n      Data content in binary mode, which means that no data conversion is performed.\\\\n      The data transfer process returns each record as-is, without translation. No\\\\n      delimiters are added between records.\\\\n\\\\n   --migrated-recall  | --mr (string)\\\\n\\\\n      The method by which migrated data set is handled. By default, a migrated data\\\\n      set is recalled synchronously. You can specify the following values: wait,\\\\n      nowait, error\\\\n\\\\n      Default value: nowait\\\\n\\\\n   --volume-serial  | --vs (string)\\\\n\\\\n      The volume serial (VOLSER) where the data set resides. You can use this option\\\\n      at any time. However, the VOLSER is required only when the data set is not\\\\n      cataloged on the system. A VOLSER is analogous to a drive name on a PC.\\\\n\\\\n ZOSMF CONNECTION OPTIONS\\\\n ------------------------\\\\n\\\\n   --host  | -H (string)\\\\n\\\\n      The z/OSMF server host name.\\\\n\\\\n   --port  | -P (number)\\\\n\\\\n      The z/OSMF server port.\\\\n\\\\n      Default value: 443\\\\n\\\\n   --user  | -u (string)\\\\n\\\\n      Mainframe (z/OSMF) user name, which can be the same as your TSO login.\\\\n\\\\n   --password  | --pass | --pw (string)\\\\n\\\\n      Mainframe (z/OSMF) password, which can be the same as your TSO password.\\\\n\\\\n   --reject-unauthorized  | --ru (boolean)\\\\n\\\\n      Reject self-signed certificates.\\\\n\\\\n      Default value: true\\\\n\\\\n   --base-path  | --bp (string)\\\\n\\\\n      The base path for your API mediation layer instance. Specify this option to\\\\n      prepend the base path to all z/OSMF resources when making REST requests. Do not\\\\n      specify this option if you are not using an API mediation layer.\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --zosmf-profile  | --zosmf-p (string)\\\\n\\\\n      The name of a (zosmf) profile to load for this command execution.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   - Upload a directory named \\\\\\"src\\\\\\" to a PDS named\\\\n   \\\\\\"ibmuser.src\\\\\\":\\\\n\\\\n      $ zowe zos-files upload dir-to-pds \\\\\\"src\\\\\\" \\\\\\"ibmuser.src\\\\\\"\\\\n\\\\n   - Upload a directory named \\\\\\"src\\\\\\" to a migrated PDS named\\\\n   \\\\\\"ibmuser.src\\\\\\" and wait for it to be recalled:\\\\n\\\\n      $ zowe zos-files upload dir-to-pds \\\\\\"src\\\\\\" \\\\\\"ibmuser.src\\\\\\" --mr wait\\\\n\\\\n\\"
}"
`;
